zlib-rs/src/c_api.rs

```diff
@@ -124,3 +124,38 @@ pub const Z_HUFFMAN_ONLY: c_int = 2;
 pub const Z_RLE: c_int = 3;
 pub const Z_FIXED: c_int = 4;
 pub const Z_DEFAULT_STRATEGY: c_int = 0;
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct gz_header {
+
+    /// true if contents is propably text
+    pub text: c_int,
+
+    /// modification time
+    pub time: z_size,
+    pub xflags: c_int,
+    pub os: c_int,
+
+    pub extra: *mut Bytef,
+    pub extra_len: uInt,
+    pub extra_max: uInt,
+
+    /// file name
+    pub name: *mut Bytef,
+    pub name_max: uInt,
+
+    /// comment
+    pub comment: *mut Bytef,
+    pub comment_max: uInt,
+
+    /// true for a header CRC16
+    pub hcrc: c_int,
+
+    /// true when done reading header
+    pub done: c_int,
+}
+
+pub type gz_headerp = *mut gz_header;
+
+pub const INFLATE_STATE_SIZE: usize = core::mem::size_of::<crate::inflate::State>();
```

Comment:
this constant is not really part of the C api, right? in any case we should preserve the comment that this should only be exposed during tests.